Assets/Scripts/EnemyPatrol.cs

```diff
@@ -0,0 +1,77 @@
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.AI;
+
+public class EnemyPatrol : MonoBehaviour
+{
+    [SerializeField] private List<GameObject> _patrolPoint;
+    [SerializeField] private List<AudioClip> _zombieAudioClip;
+    
+    private int destinationPointIndex = 0;
+    private float _minDistanceToDestination = .5f;
+    private NavMeshAgent _agent;
+    private AudioSource _zombieAudioSource;
+
+    private void Awake()
+    {
+        _agent = GetComponent<NavMeshAgent>();
+    }
+    private GameObject FindNearestPatrolPoint(List<GameObject> enemyPatrolPoints)
+    {
+        GameObject nearestPatrolPoint = null;
+        float closestDistance = Mathf.Infinity;
+        Vector3 currentPosition = transform.position;
+        foreach (GameObject potentialTarget in enemyPatrolPoints)
+        {
+            Vector3 directionToTarget = potentialTarget.transform.position - currentPosition;
+            float dSqrToTarget = directionToTarget.sqrMagnitude;
+            if (dSqrToTarget < closestDistance)
+            {
+                closestDistance = dSqrToTarget;
+                nearestPatrolPoint = potentialTarget;
+                destinationPointIndex = enemyPatrolPoints.IndexOf(potentialTarget);
+            }
+        }
+        return nearestPatrolPoint;
+    }
+
+    private void Start()
+    {
+        NavMeshAgent agent = GetComponent<NavMeshAgent>();
+        _zombieAudioSource = GetComponent<AudioSource>();
+        agent.destination = FindNearestPatrolPoint(_patrolPoint).transform.position;
```

Comment:
in line 42 you say destination is nearest patrol point, also in line 43 you say go to nearest point, which does exactly the same. please set a initial patrol point instead of calculating it, since you dont use it anywhere else and its too much useless details